pkg/controllers/dashboard/systemcharts/controller_test.go

```diff
@@ -6,43 +6,91 @@ import (
 
 	"github.com/golang/mock/gomock"
 	catalog "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
 	"github.com/rancher/rancher/pkg/controllers/dashboard/chart"
 	chartfake "github.com/rancher/rancher/pkg/controllers/dashboard/chart/fake"
 	"github.com/rancher/rancher/pkg/features"
 	"github.com/rancher/rancher/pkg/namespace"
 	"github.com/rancher/rancher/pkg/settings"
 	"github.com/rancher/wrangler/pkg/generic/fake"
+	"github.com/rancher/wrangler/pkg/relatedresource"
+	"github.com/stretchr/testify/require"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 var (
-	errUnimplemented  = fmt.Errorf("unimplemented")
-	errNotFound       = fmt.Errorf("not found")
+	errTest           = fmt.Errorf("test error")
 	priorityClassName = "rancher-critical"
 	operatorNamespace = "rancher-operator-system"
+	priorityConfig    = &v1.ConfigMap{
+		Data: map[string]string{
+			"priorityClassName": priorityClassName,
+		},
+	}
+	fullConfig = &v1.ConfigMap{
+		Data: map[string]string{
+			"priorityClassName":    priorityClassName,
+			chart.WebhookChartName: testYAML,
+		},
+	}
+	invalidConfig = &v1.ConfigMap{
+		Data: map[string]string{
+			chart.WebhookChartName: "--- %{}---\n:",
+		},
+	}
+	emptyConfig        = &v1.ConfigMap{}
+	originalMinVersion = settings.RancherWebhookMinVersion.Get()
+	originalVersion    = settings.RancherWebhookVersion.Get()
 )
 
-// Test_ChartInstallation test that all expected charts are installed or uninstalled with expected configuration
+const testYAML = `---
+newKey: newValue
+mcm:
+  enabled: false
+global: ""
+priorityClassName: newClass
+`
+
+type testMocks struct {
+	manager       *chartfake.MockManager
+	namespaceCtrl *fake.MockNonNamespacedControllerInterface[*v1.Namespace, *v1.NamespaceList]
+	configCache   *fake.MockCacheInterface[*v1.ConfigMap]
+}
+
+func (t *testMocks) Handler() *handler {
+	return &handler{
+		manager:      t.manager,
+		namespaces:   t.namespaceCtrl,
+		chartsConfig: chart.RancherConfigGetter{ConfigCache: t.configCache},
+	}
+}
+
+// Test_ChartInstallation test that all expected charts are installed or uninstalled with expected configuration.
 func Test_ChartInstallation(t *testing.T) {
 	repo := &catalog.ClusterRepo{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: repoName,
 		},
 	}
+	origMCM := features.MCM.Enabled()
+	defer func() {
```

Comment:
Nit: Personally speaking, I would do this in the specific test case that uses the value rather than here, since you are relying on the other tests referring to the current state of the feature (which may have been mutated by other test states) correctly.